docs/rst/tutorials/micro/custom_transports/custom_transports.rst

```diff
@@ -0,0 +1,279 @@
+
+In general, the members of a ``eprosima::uxr::CustomEndPoint`` object can be unsigned integers and strings.
+
+``CustomEndPoint`` defines three methods:
+
+**Add member**
+
+.. code-block:: c
+
+    bool eprosima::uxr::CustomEndPoint::add_member<*KIND*>(const std::string& member_name);
+
+This function allows to dynamically add a new member to the endpoint definition.
+
+Ir returns ``true`` if the member was correctly added, ``false`` if something went wrong (for example, if the member already exists).
+
+- **KIND**: To be chosen from: ``uint8_t``, ``uint16_t``, ``uint32_t``, ``uint64_t``, ``uint128_t`` or ``std::string``.
+- **member_name**: The tag used to identify the endpoint member.
+
+**Set member value**
+
+.. code-block:: c
+
+    void eprosima::uxr::CustomEndPoint::set_member_value(const std::string& member_name, const *KIND* & value);
+
+
+This function sets the specific value (numeric or string) for a certain member, which must previously exist in the ``CustomEndPoint``.
+
+- **member_name**: The member whose value is going to be modified.
+- **value**: The value to be set, of ``KIND``: ``uint8_t``, ``uint16_t``, ``uint32_t``, ``uint64_t``, ``uint128_t`` or ``std::string``.
+
+**Get member**
+
+.. code-block:: c
+
+    const *KIND* & eprosima::uxr::CustomEndPoint::get_member(const std::string& member_name);
+
+
+This function gets the current value of the member registered with the given parameter.
+
+The retrieved value might be an ``uint8_t``, ``uint16_t``, ``uint32_t``, ``uint64_t``, ``uint128_t`` or ``std::string``.
+
+- **member_name**: The `CustomEndPoint` member name whose current value is requested.
+
+Open function
+^^^^^^^^^^^^^
+
+.. code-block:: c
+
+    eprosima::uxr::CustomAgent::InitFunction my_custom_transport_open = [&]() -> bool
+    {
+        ...
+    }
+
+This function should open and init the custom transport. It returns a boolean indicating if the opening was successful.
+
+Close function
+^^^^^^^^^^^^^^
+
+.. code-block:: c
+
+    eprosima::uxr::CustomAgent::FiniFunction my_custom_transport_close = [&]() -> bool
+    {
+        ...
+    }
+
+This function should close the custom transport. It returns a boolean indicating if the closing was successful.
+
+Write function
+^^^^^^^^^^^^^^
+
+.. code-block:: c
+
+    eprosima::uxr::CustomAgent::SendMsgFunction my_custom_transport_write = [&](
+        const eprosima::uxr::CustomEndPoint* destination_endpoint,
+        uint8_t* buffer,
+        size_t length,
+        eprosima::uxr::TransportRc& transport_rc) -> ssize_t
+    {
+        ...
+    }
+
+This function should write data to the custom transport. It must use the ``destination_endpoint`` members to set the data destination. It returns the number of bytes written.
+
+It should set ``transport_rc`` indicating the result of the operation.
```

Comment:
Please rephrase this in one phrase or a bullet point list to avoid the `It` repetition.